### PR TITLE
Support language variants for summary and mobile content

### PR DESCRIPTION
**Pull request by Pchelolo**
cc @wikimedia/services @berndsi

### CODE REVIEW

File: v1/summary.yaml [hunk 1]
```diff
@@ -145,15 +145,16 @@ paths:
             request:
               method: put
               uri: /{domain}/sys/key_value/page_summary/{request.params.title}
-              headers: '{{merge({"if-none-hash-match": "*"}, extract.headers)}}'
+              headers: '{{merge({"if-none-hash-match": "*"}, extract.headers, "cache-control": request.headers.cache-control)}}'
```
Comment (d00rman) [inline v1/summary.yaml]: Why is the request's `cache-control` being added here and below?

Comment (Pchelolo) [reply v1/summary.yaml]: @d00rman to pass through the `no-store` header.

**Comment by Pchelolo** [comment]
Doesn't Varnish strip those headers?

**Comment by Pchelolo** [comment]
Rebased

**Comment by Pchelolo** [comment]
All the dependencies were deployed, so we're ready to go fowrward with this.

**Comment by d00rman** [comment]
> Doesn't Varnish strip those headers?

Yes, Varnish does, but what about internal clients? Better safe than sorry

File: v1/summary.yaml [hunk 2]
```diff
@@ -145,15 +145,16 @@ paths:
...
               body: '{{extract.body}}'
             # With the if-none-hash-match header the storage will return 412
             # if the content is not changed. In that case, return from the
             # handler completely, and avoid issuing purges.
+            # Also if we didn't really store anything, return as well.
             catch:
-              status: 412
+              status: [ 412, 202 ]
```
Comment (d00rman) [inline v1/summary.yaml]: Nit: let's sort these numbers.

File: v1/summary.yaml [hunk 3]
```diff
@@ -145,15 +145,16 @@ paths:
...
             return_if:
-              status: 412
+              status: [ 412, 202 ]
```
Comment (d00rman) [inline v1/summary.yaml]: Idem.

File: v1/summary.yaml (hunk 2 above)
Comment (Pchelolo) [reply v1/summary.yaml]: Done

File: lib/normalize_headers_filter.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+const mwUtil = require('./mwUtil');
+
+module.exports = function normalizeHeaders(hyper, req, next) {
+    if (mwUtil.isNoCacheRequest(req)) {
+        req.headers['cache-control'] = 'no-cache';
+    } else {
```
Comment (d00rman) [inline lib/normalize_headers_filter.js]: Nit: `else if (req.headers && req.headers['cache-control'])` (I wonder if this is needed, i.e. is there somewhere where we ensure the existence of the headers hash?)

**Pull request merged**